src/util/arith.h

```diff
@@ -0,0 +1,58 @@
+/*********************                                                        */
```

Comment:
Should we just call this file index_root_predicate.h?

Reply:
Yes, we should probably do that.